vignettes/s3-vector.Rmd

```diff
@@ -944,20 +944,44 @@ Let's assume that the `percent` class is implemented in the pizza package in the
 
 ### Getting started
 
-First, the pizza package needs to include vctrs in the `Imports` section of its `DESCRIPTION` (perhaps by calling`usethis::use_package("vctrs")`. Then we make all vctrs functions available within the pizza package by including the directive `#' @import vctrs` somewhere. You can use `usethis::use_package_doc()` to create package-level documentation in `R/pizza-doc.R`, in which case that is also the logical place to put `#' @import vctrs`. Usually, it's not good practice to `@import` the entire namespace of a package, but vctrs is deliberately designed with this use case in mind.
+First, the pizza package needs to include vctrs in the `Imports` section of its `DESCRIPTION` (perhaps by calling`usethis::use_package("vctrs")`. While vctrs is under very active development, it probably makes sense to state a minimum version.
 
-Remember to call `devtools::document()` regularly, as you develop, to regenerate `NAMESPACE` and the `.Rd` files.
+```
+Imports: 
+    a_package,
+    another_package,
+    ...
+    vctrs (>= x.y.z),
+    ...
+```
 
-Next we make a place to dump some internal documentation that's needed to avoid `R CMD check` complaints. We don't expect any human to ever read this documentation. This, and all subsequent code, presumably lives in `R/percent.R`:
+Then we make all vctrs functions available within the pizza package by including the directive `#' @import vctrs` somewhere. Usually, it's not good practice to `@import` the entire namespace of a package, but vctrs is deliberately designed with this use case in mind.
```

Comment:
I'm not sure about this advice. I think it makes sense for packages that heavily use vctrs to manipulate vectors (like tidyr), but not for packages that just implement a vctrs class (like pizza).

Reply:
A somewhat awkward thing you have to do if you are implementing a vctrs class and you don’t import all of vctrs is to individually import `vec_ptype2.<foo>` methods like `vec_ptype2.factor` here https://github.com/tidymodels/probably/blob/389d3a576237d493d98c8d50f3d92e68893ae1b6/R/vctrs-compat.R#L256

Reply:
You'll also need to do that if your class is coercible with another foreign class.

Reply:
Resolved via local discussion